src/coord/rmr/redis_cluster.c

```diff
@@ -130,13 +130,23 @@ void UpdateTopology(RedisModuleCtx *ctx) {
 }
 
 #define REFRESH_PERIOD 1000 // 1 second
+RedisModuleTimerID topologyRefreshTimer = 0;
 
 static void UpdateTopology_Periodic(RedisModuleCtx *ctx, void *p) {
   REDISMODULE_NOT_USED(p);
-  RedisModule_CreateTimer(ctx, REFRESH_PERIOD, UpdateTopology_Periodic, NULL);
+  topologyRefreshTimer = RedisModule_CreateTimer(ctx, REFRESH_PERIOD, UpdateTopology_Periodic, NULL);
   UpdateTopology(ctx);
 }
 
-void InitRedisTopologyUpdater(RedisModuleCtx *ctx) {
-  RedisModule_CreateTimer(ctx, REFRESH_PERIOD, UpdateTopology_Periodic, NULL);
+int InitRedisTopologyUpdater(RedisModuleCtx *ctx) {
+  if (topologyRefreshTimer) return REDISMODULE_ERR;
+  topologyRefreshTimer = RedisModule_CreateTimer(ctx, REFRESH_PERIOD, UpdateTopology_Periodic, NULL);
+  return REDISMODULE_OK;
+}
+
+int StopRedisTopologyUpdater(RedisModuleCtx *ctx) {
+  if (clusterConfig.type != ClusterType_RedisOSS) return REDISMODULE_OK; // Only Redis OSS supports this
```

Comment:
nit: don't you think it's better to return an error if the cluster is not OSS? Also, should we validate the cluster type upon resume as well?

Reply:
1. The reason I chose `OK` is in case we have a generic flow for loss and enterprise (maybe even in a test) or if someone calls it manually in an enterprise environment. Since I ended up returning an error and not asserting in case of an error, we can return an error here.
2. The "resume" function is the original init function that is being used in OSS-only code. But I can add a validation there as well